src/parsers/manifest/dash/parse_periods.ts

```diff
@@ -45,44 +55,177 @@ export default function parsePeriods(
-    const period = periodsIR[i];
+  // We parse it in reverse because we might need to deduce the buffer depth from
+  // the last Periods' indexes
+  const { manifestBoundsCalculator } = manifestInfos;
+  for (let i = periodsIR.length - 1; i >= 0; i--) {
+    const periodIR = periodsIR[i];
+    const periodBaseURL = resolveURL(manifestInfos.baseURL, periodIR.children.baseURL);
+
     const { periodStart,
             periodDuration,
             periodEnd } = periodsTimeInformations[i];
 
-    const periodBaseURL = resolveURL(manifestInfos.baseURL, period.children.baseURL);
-
     let periodID : string;
-    if (period.attributes.id == null) {
+    if (periodIR.attributes.id == null) {
       log.warn("DASH: No usable id found in the Period. Generating one.");
       periodID = "gen-dash-period-" + generatePeriodID();
     } else {
-      periodID = period.attributes.id;
+      periodID = periodIR.attributes.id;
     }
 
     const periodInfos = { availabilityStartTime: manifestInfos.availabilityStartTime,
                           baseURL: periodBaseURL,
+                          manifestBoundsCalculator:
+                            manifestInfos.manifestBoundsCalculator,
                           clockOffset: manifestInfos.clockOffset,
                           end: periodEnd,
                           isDynamic: manifestInfos.isDynamic,
                           start: periodStart,
                           timeShiftBufferDepth: manifestInfos.timeShiftBufferDepth };
-    const adaptations = parseAdaptationSets(period.children.adaptations,
+    const adaptations = parseAdaptationSets(periodIR.children.adaptations,
                                             periodInfos);
     const parsedPeriod : IParsedPeriod = { id: periodID,
                                            start: periodStart,
                                            end: periodEnd,
                                            duration: periodDuration,
                                            adaptations };
-    parsedPeriods.push(parsedPeriod);
+    parsedPeriods.unshift(parsedPeriod);
+
+    if (manifestInfos.isDynamic && !manifestBoundsCalculator.lastPositionIsKnown()) {
+      if (manifestInfos.isDynamic) {
+      // Try to guess last position to obtain the buffer depth
+      const lastPosition = getMaximumLastPosition(adaptations);
+      if (typeof lastPosition === "number") { // last position is available
+        const clockOffset = performance.now() / 1000;
+        manifestBoundsCalculator.setLastPosition(lastPosition, clockOffset);
+      } else {
+        const [guessedLastPosition, guessedClockOffset] =
+          guessLastPositionOffsetFromClock(manifestInfos, periodStart);
+        if (guessedLastPosition != null && guessedClockOffset != null) {
+          manifestBoundsCalculator.setLastPosition(
+            guessedLastPosition, guessedClockOffset);
+        }
+      }
+      } else if (manifestInfos.duration != null) {
```

Comment:
braces don't work man

Reply:
Also might be better if that condition was much higher (it is a single line, it would be easier to follow)

Reply:
Or even better, just do that before parsing the last Period, make little sense to put it here if it does not depend on a parsed period.